### PR TITLE
ACP 118 reference implementation

### PR DESCRIPTION
**Pull request by cam-schultz**
## Why this should be merged
- Adds proto definitions for `SignatureRequest` and `SignatureResponse` `AppRequest`/`AppResponse` message types as defined in [ACP-118](https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/118-warp-signature-request)
- ~~Defines the canonical handler ID for `SignatureRequests` as `1`, as specified in https://github.com/avalanche-foundation/ACPs/pull/129~~
- Defines the canonical handler ID for `SignatureRequests` as `2`, as specified in https://github.com/avalanche-foundation/ACPs/pull/132
- Moves the existing definition of the handler ID for Tx Gossip messages to the `p2p` package

Does **not** go so far as to add a new package under `network/p2p` with message parsing, message marshalling, default handlers, etc. ala the [gossip](https://github.com/ava-labs/avalanchego/tree/master/network/p2p/gossip) package.

## How this works
See above

## How this was tested
CI

### CODE REVIEW

**Review by michaelkaplan13 — APPROVED** [review]
Makes sense to me, would defer to those with more context of the codebase to comment on where the definitions live within the files/packages.

**Review by tsachiherman — APPROVED** [review]
looks good to me, but would be happy to have others review this as well.

File: network/p2p/handler.go [hunk 1]
```diff
@@ -21,6 +21,11 @@ var (
 	_ Handler = (*ValidatorHandler)(nil)
 )
 
+const (
+	TxGossipHandlerID = iota
+	SignatureRequestHandlerID
+)
```
Comment (StephenButtolph) [inline network/p2p/handler.go:27]: Super nit - but typically we define `const` blocks before `var` blocks

File: network/p2p/handler.go [hunk 2]
```diff
@@ -21,6 +21,11 @@ var (
 	_ Handler = (*ValidatorHandler)(nil)
 )
 
+const (
+	TxGossipHandlerID = iota
+	SignatureRequestHandlerID
```
Comment (StephenButtolph) [inline network/p2p/handler.go:26]: This actually conflicts with the existing `atomicTxGossipHandlerID` used by `coreth`. We should probably update this and the ACP accordingly.

Comment (cam-schultz) [reply network/p2p/handler.go]: Bumped to `2` by adding `AtomicTxGossipHandlerID` here. Corresponding ACP change: https://github.com/avalanche-foundation/ACPs/pull/132

File: proto/sdk/sdk.proto [hunk 3]
```diff
@@ -16,3 +16,12 @@ message PullGossipResponse {
...
 }
+
+message SignatureRequest {
+  bytes message = 1;
+  bytes justification = 2;
+}
+
+message SignatureResponse {
+  bytes signature = 1;
+}
```
Comment (StephenButtolph) [inline proto/sdk/sdk.proto:27]: Can document these with links out to the ACP?

File: network/p2p/handler.go [hunk 4]
```diff
@@ -15,6 +15,12 @@ import (
 	"github.com/ava-labs/avalanchego/utils/logging"
 )
 
+const (
+	TxGossipHandlerID = iota
+	AtomicTxGossipHandlerID
+	SignatureRequestHandlerID
```
Comment (StephenButtolph) [inline network/p2p/handler.go:21]: Can we document these as standards and link out to the ACP in a comment?

**Comment by StephenButtolph** [comment]
After adding the couple requested comments - lgtm

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**